pootle/apps/import_export/utils.py

```diff
@@ -12,10 +12,14 @@
 from django.utils.translation import ugettext as _
 
 from pootle_store.models import Store
+from pootle_statistics.models import SubmissionTypes
 
 
-def import_file(file):
+def import_file(file, user=None):
     f = getclass(file)(file.read())
+    if not hasattr(f, "parseheader"):
+        raise ValueError(_("Unsupported filetype '%s', only po files " % (file.name)
+                           + "are supported at this time\n"))
```

Comment:
get_class has responsibility for parsing the mimetype - file.name is just used to indicate to the user the name of the file that they have uploaded